Added webcam -> cvResize -> cvtColor(BGR2BGRA)

Created the pipeline webcam -> cvResize -> cvtColor(BGR2BGRA), and it crashed.

I can confirm this on mac. It took the entire build down, I didn't get the "this exception was thrown" screen, so that needs to be fixed first. Here's the error, from the console:

```
java.lang.AssertionError: Only 3-channel BGR images or single-channel grayscale images can be converted
        at edu.wpi.grip.ui.util.ImageConverter.convert(ImageConverter.java:55)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView.lambda$null$0(ImageSocketPreviewView.java:31)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView$$Lambda$680/778946230.run(Unknown Source)
        at edu.wpi.grip.ui.util.GripPlatform.runAsSoonAsPossible(GripPlatform.java:51)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView.lambda$convertImage$1(ImageSocketPreviewView.java:30)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView$$Lambda$679/201664973.accept(Unknown Source)
        at java.util.Optional.ifPresent(Optional.java:159)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView.convertImage(ImageSocketPreviewView.java:29)
        at edu.wpi.grip.ui.preview.ImageBasedPreviewView.resize(ImageBasedPreviewView.java:63)
        at edu.wpi.grip.ui.preview.PreviewsController.lambda$onSocketPreviewChanged$8(PreviewsController.java:134)
        at edu.wpi.grip.ui.preview.PreviewsController$$Lambda$678/459878562.run(Unknown Source)
        at com.sun.javafx.application.PlatformImpl.runAndWait(PlatformImpl.java:317)
        at com.sun.javafx.application.PlatformImpl.runAndWait(PlatformImpl.java:307)
        at edu.wpi.grip.ui.preview.PreviewsController.onSocketPreviewChanged(PreviewsController.java:128)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
        at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
        at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
        at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:435)
        at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
        at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
        at com.google.common.eventbus.EventBus.post(EventBus.java:217)
        at edu.wpi.grip.core.sockets.OutputSocketImpl.setPreviewed(OutputSocketImpl.java:45)
        at edu.wpi.grip.ui.pipeline.OutputSocketController.lambda$new$0(OutputSocketController.java:61)
        at edu.wpi.grip.ui.pipeline.OutputSocketController$$Lambda$576/1363243807.invalidated(Unknown Source)
        at com.sun.javafx.binding.ExpressionHelper$SingleInvalidation.fireValueChangedEvent(ExpressionHelper.java:137)
        at com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:81)
        at javafx.beans.property.BooleanPropertyBase.fireValueChangedEvent(BooleanPropertyBase.java:103)
        at javafx.beans.property.BooleanPropertyBase.markInvalid(BooleanPropertyBase.java:110)
        at javafx.beans.property.BooleanPropertyBase.set(BooleanPropertyBase.java:144)
        at javafx.scene.control.ToggleButton.setSelected(ToggleButton.java:149)
        at javafx.scene.control.ToggleButton.fire(ToggleButton.java:255)
        at com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:182)
        at com.sun.javafx.scene.control.skin.BehaviorSkinBase$1.handle(BehaviorSkinBase.java:96)
        at com.sun.javafx.scene.control.skin.BehaviorSkinBase$1.handle(BehaviorSkinBase.java:89)
        at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
        at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
        at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
        at javafx.event.Event.fireEvent(Event.java:198)
        at javafx.scene.Scene$MouseHandler.process(Scene.java:3758)
        at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3486)
        at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
        at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2495)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:350)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:275)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$350(GlassViewEventHandler.java:385)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$$Lambda$476/352098194.get(Unknown Source)
        at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:404)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:384)
        at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
        at com.sun.glass.ui.View.notifyMouse(View.java:927)
Jan 02, 2017 6:39:00 PM edu.wpi.grip.core.GripCoreModule onThreadException
SEVERE: Uncaught Exception on thread Thread[JavaFX Application Thread,5,main]
java.lang.AssertionError: Only 3-channel BGR images or single-channel grayscale images can be converted
        at edu.wpi.grip.ui.util.ImageConverter.convert(ImageConverter.java:55)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView.lambda$null$0(ImageSocketPreviewView.java:31)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView$$Lambda$680/778946230.run(Unknown Source)
        at edu.wpi.grip.ui.util.GripPlatform.lambda$onJavaFXRunnerEvent$0(GripPlatform.java:74)
        at edu.wpi.grip.ui.util.GripPlatform$$Lambda$681/739159934.run(Unknown Source)
        at com.sun.javafx.application.PlatformImpl.lambda$null$170(PlatformImpl.java:295)
        at com.sun.javafx.application.PlatformImpl$$Lambda$49/1430038102.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$171(PlatformImpl.java:294)
        at com.sun.javafx.application.PlatformImpl$$Lambda$48/237061348.run(Unknown Source)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)

Jan 02, 2017 6:39:00 PM edu.wpi.grip.core.GripCoreModule onThreadException
SEVERE: Uncaught Exception on thread Thread[JavaFX Application Thread,5,main]
java.lang.AssertionError: Only 3-channel BGR images or single-channel grayscale images can be converted
        at edu.wpi.grip.ui.util.ImageConverter.convert(ImageConverter.java:55)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView.lambda$null$0(ImageSocketPreviewView.java:31)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView$$Lambda$680/778946230.run(Unknown Source)
        at edu.wpi.grip.ui.util.GripPlatform.lambda$onJavaFXRunnerEvent$0(GripPlatform.java:74)
        at edu.wpi.grip.ui.util.GripPlatform$$Lambda$681/739159934.run(Unknown Source)
        at com.sun.javafx.application.PlatformImpl.lambda$null$170(PlatformImpl.java:295)
        at com.sun.javafx.application.PlatformImpl$$Lambda$49/1430038102.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$171(PlatformImpl.java:294)
        at com.sun.javafx.application.PlatformImpl$$Lambda$48/237061348.run(Unknown Source)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.mac.MacApplication._enterNestedEventLoopImpl(Native Method)
        at com.sun.glass.ui.mac.MacApplication._enterNestedEventLoop(MacApplication.java:109)
        at com.sun.glass.ui.Application.enterNestedEventLoop(Application.java:511)
        at com.sun.glass.ui.EventLoop.enter(EventLoop.java:107)
        at com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:593)
        at javafx.stage.Stage.showAndWait(Stage.java:463)
        at javafx.scene.control.HeavyweightDialog.showAndWait(HeavyweightDialog.java:157)
        at javafx.scene.control.Dialog.showAndWait(Dialog.java:313)
        at edu.wpi.grip.ui.Main.lambda$null$4(Main.java:188)
        at edu.wpi.grip.ui.Main$$Lambda$683/787628229.run(Unknown Source)
        at com.sun.javafx.application.PlatformImpl.runAndWait(PlatformImpl.java:317)
        at com.sun.javafx.application.PlatformImpl.runAndWait(PlatformImpl.java:307)
        at edu.wpi.grip.ui.Main.lambda$onUnexpectedThrowableEvent$5(Main.java:177)
        at edu.wpi.grip.ui.Main$$Lambda$682/36711416.handle(Unknown Source)
        at edu.wpi.grip.core.events.UnexpectedThrowableEvent.handleSafely(UnexpectedThrowableEvent.java:51)
        at edu.wpi.grip.ui.Main.onUnexpectedThrowableEvent(Main.java:172)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
        at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
        at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
        at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:435)
        at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
        at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
        at com.google.common.eventbus.EventBus.post(EventBus.java:217)
        at edu.wpi.grip.core.GripCoreModule.onThreadException(GripCoreModule.java:199)
        at edu.wpi.grip.core.GripCoreModule$$Lambda$55/1861073427.uncaughtException(Unknown Source)
        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1057)
        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1052)
        at com.sun.glass.ui.Application.reportException(Application.java:453)

Jan 02, 2017 6:39:00 PM edu.wpi.grip.core.events.UnexpectedThrowableEvent shutdownIfFatal
SEVERE: Shutting down from error
java.lang.AssertionError: Only 3-channel BGR images or single-channel grayscale images can be converted
        at edu.wpi.grip.ui.util.ImageConverter.convert(ImageConverter.java:55)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView.lambda$null$0(ImageSocketPreviewView.java:31)
        at edu.wpi.grip.ui.preview.ImageSocketPreviewView$$Lambda$680/778946230.run(Unknown Source)
        at edu.wpi.grip.ui.util.GripPlatform.lambda$onJavaFXRunnerEvent$0(GripPlatform.java:74)
        at edu.wpi.grip.ui.util.GripPlatform$$Lambda$681/739159934.run(Unknown Source)
        at com.sun.javafx.application.PlatformImpl.lambda$null$170(PlatformImpl.java:295)
        at com.sun.javafx.application.PlatformImpl$$Lambda$49/1430038102.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$171(PlatformImpl.java:294)
        at com.sun.javafx.application.PlatformImpl$$Lambda$48/237061348.run(Unknown Source)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.mac.MacApplication._enterNestedEventLoopImpl(Native Method)
        at com.sun.glass.ui.mac.MacApplication._enterNestedEventLoop(MacApplication.java:109)
        at com.sun.glass.ui.Application.enterNestedEventLoop(Application.java:511)
        at com.sun.glass.ui.EventLoop.enter(EventLoop.java:107)
        at com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:593)
        at javafx.stage.Stage.showAndWait(Stage.java:463)
        at javafx.scene.control.HeavyweightDialog.showAndWait(HeavyweightDialog.java:157)
        at javafx.scene.control.Dialog.showAndWait(Dialog.java:313)
        at edu.wpi.grip.ui.Main.lambda$null$4(Main.java:188)
        at edu.wpi.grip.ui.Main$$Lambda$683/787628229.run(Unknown Source)
        at com.sun.javafx.application.PlatformImpl.runAndWait(PlatformImpl.java:317)
        at com.sun.javafx.application.PlatformImpl.runAndWait(PlatformImpl.java:307)
        at edu.wpi.grip.ui.Main.lambda$onUnexpectedThrowableEvent$5(Main.java:177)
        at edu.wpi.grip.ui.Main$$Lambda$682/36711416.handle(Unknown Source)
        at edu.wpi.grip.core.events.UnexpectedThrowableEvent.handleSafely(UnexpectedThrowableEvent.java:51)
        at edu.wpi.grip.ui.Main.onUnexpectedThrowableEvent(Main.java:172)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
        at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
        at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
        at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:435)
        at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
        at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
        at com.google.common.eventbus.EventBus.post(EventBus.java:217)
        at edu.wpi.grip.core.GripCoreModule.onThreadException(GripCoreModule.java:199)
        at edu.wpi.grip.core.GripCoreModule$$Lambda$55/1861073427.uncaughtException(Unknown Source)
        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1057)
        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1052)
        at com.sun.glass.ui.Application.reportException(Application.java:453)
```

--- a/ui/src/main/java/edu/wpi/grip/ui/pipeline/OutputSocketController.java
+++ b/ui/src/main/java/edu/wpi/grip/ui/pipeline/OutputSocketController.java
@@ -1,12 +1,14 @@
 package edu.wpi.grip.ui.pipeline;
 
 import edu.wpi.grip.core.events.BenchmarkEvent;
+import edu.wpi.grip.core.events.SocketChangedEvent;
 import edu.wpi.grip.core.events.SocketPreviewChangedEvent;
 import edu.wpi.grip.core.sockets.OutputSocket;
 import edu.wpi.grip.core.sockets.Socket;
 import edu.wpi.grip.core.sockets.SocketHint;
 import edu.wpi.grip.ui.Controller;
 import edu.wpi.grip.ui.annotations.ParametrizedController;
+import edu.wpi.grip.ui.preview.ImageBasedPreviewView;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.eventbus.Subscribe;
@@ -23,6 +25,7 @@ import javafx.scene.layout.HBox;
 import javafx.scene.layout.StackPane;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static org.bytedeco.javacpp.opencv_core.Mat;
 
 /**
  * A JavaFX control that renders an {@link OutputSocket} that is the output of a step.  It shows a
@@ -34,7 +37,7 @@ public class OutputSocketController implements Controller {
 
   private final SocketHandleView.Factory socketHandleFactory;
   private final InvalidationListener previewListener;
-  private final OutputSocket socket;
+  private final OutputSocket<?> socket;
   @FXML
   private HBox root;
   @FXML
@@ -79,12 +82,41 @@ public class OutputSocketController implements Controller {
     this.type.setText(this.socket.getSocketHint().getTypeLabel());
   }
 
-  public Socket getSocket() {
+  public Socket<?> getSocket() {
     return this.socket;
   }
 
   public SocketHandleView getHandle() {
     return this.handle;
+  }
+
+  @Subscribe
+  public void onSocketChanged(SocketChangedEvent event) {
+    if (event.isRegarding(this.socket)) {
+      if (!this.socket.getValue().isPresent()) {
+        // No value
+        handlePreview(false);
+      } else if (!(this.socket.getValue().get() instanceof Mat)) {
+        // There is a non-image value, which can always be previewed
+        handlePreview(true);
+      } else {
+        // Only allow the image to be previewed if it's previewable
+        boolean previewable = this.socket.getValue()
+            .map(Mat.class::cast)
+            .map(ImageBasedPreviewView::isPreviewable)
+            .get();
+        handlePreview(previewable);
+      }
+    }
+  }
+
+  /**
+   * Disables the preview button and hides the preview if the socket value isn't able to be
+   * previewed.
+   */
+  private void handlePreview(boolean previewable) {
+    this.preview.setDisable(!previewable);
+    this.socket.setPreviewed(this.socket.isPreviewed() && previewable);
   }
 
   @Subscribe

--- a/ui/src/main/java/edu/wpi/grip/ui/preview/ContoursSocketPreviewView.java
+++ b/ui/src/main/java/edu/wpi/grip/ui/preview/ContoursSocketPreviewView.java
@@ -3,12 +3,10 @@ package edu.wpi.grip.ui.preview;
 import edu.wpi.grip.core.operations.composite.ContoursReport;
 import edu.wpi.grip.core.sockets.OutputSocket;
 import edu.wpi.grip.ui.util.GripPlatform;
-import edu.wpi.grip.ui.util.ImageConverter;
 
 import javafx.scene.control.CheckBox;
 import javafx.scene.control.Label;
 import javafx.scene.image.Image;
-import javafx.scene.image.ImageView;
 import javafx.scene.layout.VBox;
 
 import static org.bytedeco.javacpp.opencv_core.CV_8UC3;
@@ -32,8 +30,6 @@ public final class ContoursSocketPreviewView extends ImageBasedPreviewView<Conto
       Scalar.BLUE,
       Scalar.MAGENTA,
   };
-  private final ImageConverter imageConverter = new ImageConverter();
-  private final ImageView imageView = new ImageView();
   private final Label infoLabel = new Label();
   private final CheckBox colorContours;
   private final Mat tmp = new Mat();

--- a/ui/src/main/java/edu/wpi/grip/ui/preview/ImageBasedPreviewView.java
+++ b/ui/src/main/java/edu/wpi/grip/ui/preview/ImageBasedPreviewView.java
@@ -9,6 +9,10 @@ import com.google.common.eventbus.Subscribe;
 import javafx.application.Platform;
 import javafx.scene.image.ImageView;
 
+import static org.bytedeco.javacpp.opencv_core.CV_8S;
+import static org.bytedeco.javacpp.opencv_core.CV_8U;
+import static org.bytedeco.javacpp.opencv_core.Mat;
+
 /**
  * Base class for image previews.
  */
@@ -46,6 +50,18 @@ public abstract class ImageBasedPreviewView<T> extends SocketPreviewView<T> {
    * Converts the input data to an image and render it in the {@link #imageView}.
    */
   protected abstract void convertImage();
+
+  /**
+   * Checks if an image is able to be previewed.
+   *
+   * @param image the image to check
+   *
+   * @return true if the image can be previewed, false if it can't
+   */
+  public static boolean isPreviewable(Mat image) {
+    return (image.channels() == 1) || (image.channels() == 3)
+        && (image.depth() == CV_8U || image.depth() == CV_8S);
+  }
 
   /**
    * Updates the image preview when the pipeline runs.

--- a/ui/src/main/java/edu/wpi/grip/ui/preview/ImageSocketPreviewView.java
+++ b/ui/src/main/java/edu/wpi/grip/ui/preview/ImageSocketPreviewView.java
@@ -26,12 +26,14 @@ public class ImageSocketPreviewView extends ImageBasedPreviewView<Mat> {
   @Override
   protected void convertImage() {
     synchronized (this) {
-      this.getSocket().getValue().ifPresent(mat -> {
-        platform.runAsSoonAsPossible(() -> {
-          Image image = imageConverter.convert(mat, getImageHeight());
-          imageView.setImage(image);
-        });
-      });
+      this.getSocket().getValue()
+          .filter(ImageBasedPreviewView::isPreviewable)
+          .ifPresent(mat -> {
+            platform.runAsSoonAsPossible(() -> {
+              Image image = imageConverter.convert(mat, getImageHeight());
+              imageView.setImage(image);
+            });
+          });
     }
   }
 }

